cql3/expr/expression.hh

```diff
@@ -404,12 +404,44 @@ nested_expression::nested_expression(ExpressionElement auto e)
         : nested_expression(expression(std::move(e))) {
 }
 
+// Extracts all binary operators which have the given column on their left hand side.
+// Extracts only single-column restrictions.
+// Does not include multi-column restrictions.
```

Comment:
This is fine and correct, but note that it invalidates [this comment](https://github.com/scylladb/scylla/pull/9122#issuecomment-890929644) of yours: until we can handle multi-column restrictions here, we must rely on filtering.